src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Query.vb

```diff
@@ -64,78 +100,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                        parameter.Type)
 
                     If isReservedName AndAlso Not String.Equals(parameterName, StringConstants.Group, StringComparison.Ordinal) Then
-                        ' Compound variable.
-                        ' Each range variable is an Anonymous Type property.
-                        Debug.Assert(parameterName.Equals(StringConstants.It) OrElse parameterName.Equals(StringConstants.It1) OrElse parameterName.Equals(StringConstants.It2))
-                        PopulateRangeVariableMapForAnonymousType(node.Syntax, paramRef, nodeRangeVariables, firstUnmappedRangeVariable)
+                        If parameter.Type.IsErrorType() Then
+                            ' Skip adding to the range variable map for error case.
+                            firstUnmappedRangeVariable += 1
```

Comment:
>firstUnmappedRangeVariable += 1 [](start = 28, length = 31)

It doesn't look like it is safe to assume that only one range variable should be skipped in this case. Probably we should simply exit the loop. I'd like to take a closer look at the specific scenarios, for which we are taking this code path.

Reply:
As we discussed offline, I have changed the code to exit the function at this point and also removed the assert in the rewriter.

---
In reply to: [139533872](https://github.com/dotnet/roslyn/pull/21356#discussion_r139533872) [](ancestors = 139533872)